### ISSUE TITLE
No --out file when using --file option

### ISSUE THREAD

**Issue by codeability-ab** [RSeidelsohn/license-checker-rseidelsohn#10]
Hello!
Great to see that you forked the seemingly dead license-checker project!

I'm trying to use both the `--file` and `--out` options (as well as `--json` and a list of `--excludePackages`)

There's no json file generated :-(

The json file is generated just fine if I remove the `--file` option

Any idea what's happening?

best regards Johan

**Comment by RSeidelsohn** [comment]
Hi Johan,

thanks for opening this issue! For some reason I didn't see any notification of that, so I only see it now randomly.
I will happily jump on that issue, but as this is just a side-project of mine and I have a full time job, family and other hobbies, too, I  don't work on it during the week, but usually only on week-ends (and not on every week end). So if things seem to take too long, it might help to give me a reminder. But I am really happy to work on new feature requests and make the users of this module happy! So I hope to have a look on this issue next week end!

Cheers, Roman.

**Comment by RSeidelsohn** [comment]
I just had a short look at it and it might've been an easy fix. Would you be so kind as to check out and try the [develop branch](https://github.com/RSeidelsohn/license-checker-rseidelsohn/tree/develop) and tell me if this works fine for you? That would be great. If not, please also tell me so, so that I can have a more thorough look into the issue.

Cheers, Roman.

**Comment by codeability-ab** [comment]
Hello!
It seems to work as expected!

I did this

`npm i --save-dev RSeidelsohn/license-checker-rseidelsohn#develop`

and then added

`"license-test": "license-checker-rseidelsohn --production --failOn GPL --json --files license/ --out license/licenses.json --excludePackages @ournamespace/foo;@ournamespace/bar",
`
to package.json

and the command

`npm run license-test`

results in tons of license files and a licenses.json in the license/ directory

:-)

/Johan

**Comment by codeability-ab** [comment]
Hmm... should I expect the licenses.json file to have updated licenseFile paths and filenames?

The licenseFile fields still point to the node_modules directory and the original filenames

/Johan

**Comment by RSeidelsohn** [comment]
I'll have a look soon, thx for the update!

**Comment by RSeidelsohn** [comment]
Hi @codeability-ab ,

I just checked the setup. What would you expect the `licenseFile` fields to point to?
In fact I checked the setup with adding the `--relativeModulePath` option (which was the inital reason for me to work on the original `license-checker`) and saw that this does not work with the `--out`-option. So I will hopefully fix that now, but I still wonder what you would expect for the filenames - should the combination of `--out` and `--files` make the `licenseFile` paths point to the files just created in the (in your example) `/licenses` directory? I would not expect that, tbh. But I could possibly create a new option for that functionality.

Cheers, Roman.

**Comment by RSeidelsohn** [comment]
I will close this issue if until 1st of july I will get no response from you, @codeability-ab - I hope that you are ok with this decision. If you read this after that date, please feel free to re-open the issue!

Cheers, Roman.

**Comment by codeability-ab** [comment]
Hello!
Sorry for my late response!

I would expect the paths to the license files to be relative to the generated licenses.json file when using --files and --out

When using --files each file is copied and given a consistent name which is great. The path and new filename really should be added to the license entry in the --out file

If I run this command
```
license-checker-rseidelsohn --production --json --out licenses/licenses.json --files licenses/files/
```
then this would be pretty good
```
  "react@16.14.0": {
    "licenses": "MIT",
    "repository": "https://github.com/facebook/react",
    "path": "/Users/johan/projects/my-project/node_modules/react",
    "licenseFile": "./files/react@16.14.0-LICENSE.txt"
  }, 
  ...
```

We're using your license checker software to let our build server generate an info page listing third party software for our web application which is also distributed as Cordova packaged iOS and Android apps - https://login.ecster.se/#/about/license

Right now we're running a post process script to fix the path issue.

Thanks for your work!

best regards Johan

**Comment by RSeidelsohn** [comment]
Ok, @codeability-ab, skiing holidays is a good time for me to work on spare time projects, as I do not ski myself and the rest of the family is out of the house. So I took the chance and released [version 3.0.0](https://github.com/RSeidelsohn/license-checker-rseidelsohn/releases/tag/v3.0.0) that offers what you wanted. I changed your request a bit in the respect that you need to ue the already existing `--relativeLicensePath` option if you want the licenseFile paths to be relative to the `--out` directory in case you combine these options. If you leave the `relativeLicensePath` option, then the absolute paths to the copied license files will be used.
Please feel free to give it a try and tell me if this works fine for you.

Cheers, Roman.

**Comment by RSeidelsohn** [comment]
Hi @denniseffing ,

thanks a lot for your efforts and the good description. This will be the next bug I'll fix. My regular work on this project will start in about two weeks, and every 2nd friday from then on.

Cheers, Roman.

**Comment by RSeidelsohn** [comment]
Dear @denniseffing and @codeability-ab ,

as I just saw how easy this bug was to fix, I released a new bugfix version 4.2.2 that now produces the correct relative paths for this combination of options. Sorry that it took that long. If you do not object in the next days, I will close this issue as fixed.

Cheers, Roman.
